### ISSUE TITLE
Diff is not passed through from inner assertion

### ISSUE THREAD

**Issue by gertsonderby** [alexjeffburke/unexpected-mxhr#1]
Using the following simple wrapper function on fetch (Using the whatwg-fetch polyfill):
```js
export default function requestJSON(inputData) {
	return fetch(inputData)
	.then(response => response.json());
}
```
The following test fails, as expected:
```js
// Inside describe('Access requestJSON', ...)
it('is asynchronous', () =>
	expect(() =>
		requestJSON('/')
		.then(result => expect(result, 'to be a string')), // This should be an object
	'with xhr mocked out', {
		request: 'GET /',
		response: { body: '{}' }
	},
	'not to error')
);
```
However, nothing is reported about the inner, failed assertion. This is the full output:
```
  1) Access requestJSON is asynchronous:

expected
function () {
      return (0, _requestJSON2.default)('/').then(function (result) {
        return (0, _expect2.default)(result, 'to be a string');
      });
    }
with xhr mocked out { request: 'GET /', response: { body: '{}' } } not to error
```
Replacing `not to error` with a custom assertion and changing error modes seems to have no effect. Trying to access the assertion error's parentage via `withError()` also produces nothing.

**Comment by alexjeffburke** [comment]
Oh this is something we've not got a lot of experience dealing with in -mitm - the implementation is a little different but I've got a good idea where to look :) Give me a day or two at most and I'll get you an update.

**Comment by alexjeffburke** [comment]
Quick update - I'm hoping the changes in 0.2.1 help this situation. We've added a test case that has a delegated rejection from `not to error` that checks it is reported :)

**Comment by gustavnikolaj** [comment]
Just had a similar case to this that @papandreou and I fixed on #3
